File: frontend/movies/src/types/movie.ts
export interface Movie {
  id: number;
  title: string;
  overview: string;
  imagePath: string;
  certificate: string;
  director: string;
  releaseDate: number;
  runtime: number;
  numberOfVotes: number;
  revenue: number;
  metaScore: number;
  imdbRating: number;
  genres: { name: string }[];
  stars: { name: string }[];
}

export interface MovieResponse {
  content: Movie[];
  pageable: {
    pageNumber: number;
    pageSize: number;
  };
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
  last: boolean;
  first: boolean;
}

File: backend/movies/src/main/java/elte/hu/movies/entity/Movie.java
package elte.hu.movies.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;
import java.util.stream.Stream;

@Entity
@Table(name="movies")
@AllArgsConstructor
@RequiredArgsConstructor
@Builder
public class Movie {

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    @Getter
    Long id;

    @Getter
    @Setter
    String title;

    @Getter
    @Setter
    @Column(length = 2000)
    String overview;

    @Getter
    @Setter
    String imagePath;

    @Getter
    @Setter
    String certificate;

    @Getter
    @Setter
    String director;

    @Getter
    @Setter
    int releaseDate;

    @Getter
    @Setter
    int runtime;

    @Getter
    @Setter
    int numberOfVotes;

    @Getter
    @Setter
    Long revenue;

    @Getter
    @Setter
    Double metaScore;

    @Getter
    @Setter
    double imdbRating;

    @Setter
    @ManyToMany
    @JoinTable(
            name = "movies_genres",
            joinColumns = @JoinColumn(name = "movie_id"),
            inverseJoinColumns = @JoinColumn(name = "genre_name"))
    List<Genre> genres;

    @Setter
    @ManyToMany
    @JoinTable(
            name = "movies_stars",
            joinColumns = @JoinColumn(name = "movie_id"),
            inverseJoinColumns = @JoinColumn(name = "star_name"))
    List<Star> stars;


    public Stream<Genre> getGenres() {
        return genres.stream();
    }

    public Stream<Star> getStars() {
        return stars.stream();
    }

}
